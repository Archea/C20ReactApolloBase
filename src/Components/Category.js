import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router'
import { ApolloConsumer } from 'react-apollo'
/*
  renders the category based on the current route
*/
export class Category extends Component {
  //thing = <span onClick={() => client.writeData({ data: { category.selected: this.props.Id } })}></span>
  clickSelect = id => (
    <ApolloConsumer>
      {client => (
        <div
          onClick={() =>
            client.writeData({
              data: { category: { __typename: 'Category', selected: id } }
            })
          }>
          select
        </div>
      )}
    </ApolloConsumer>
  )

  render() {
    return (
      <div className="Category">
        {this.props.Category.name}
        {this.props.Selected ? (
          <div>ooOOOoo</div>
        ) : (
          this.clickSelect(this.props.Id)
        )}
      </div>
    )
  }
}

export default withRouter(Category)
