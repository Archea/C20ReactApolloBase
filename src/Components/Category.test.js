import React from 'react'
import ReactDOM from 'react-dom'
import { Category, TOGGLE_SELECTED } from './Category'
import renderer from 'react-test-renderer'
import { StaticRouter } from 'react-router'
import { MockedProvider } from 'react-apollo/test-utils'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe('Category', () => {
  const mocks = {
    request: {
      query: TOGGLE_SELECTED,
      variables: {
        id: '1'
      }
    },
    result: {
      data: {
        Category: {
          id: '1',
          name: 'bar'
        }
      }
    }
  }
  const props = {
    Category: {
      id: '1',
      name: 'bar'
    }
  }
  test.skip('it at renders and matches snapshot', () => {
    const context = {}
    const comp = shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Category {...props} />
      </MockedProvider>
    )
    expect(comp).toBeDefined()
    expect(comp.toJSON()).toMatchSnapshot()
  })
})
