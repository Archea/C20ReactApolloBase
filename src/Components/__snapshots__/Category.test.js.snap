// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category it at renders and matches snapshot 1`] = `
<div
  className="Category"
>
  bar
</div>
`;
